onelogin_rules/onelogin_brute_force_by_ip.py

```diff
@@ -0,0 +1,12 @@
+def rule(event):
+    # filter events; event type 6 is a failed authentication
+    return event.get('event_type_id') == 6
+
+
+def dedup(event):
+    return event.get('ipaddr', '<UNKNOWN_IP>')
+
+
+def title(event):
+    return 'IP [{}] has exceeded the failed logins threshold'.format(
+        event.get('ipaddr'))
```

Comment:
If `dedup` is not defined, we default to using the value of the `title` function instead. In cases where the `dedup` and `title` function are always relying on the same fields from the event (in this case `ipaddr`), we try to skip the `dedup` function.

So in this case I would drop the `dedup` function entirely, and add the `<UNKNOWN_IP>` as a default value in the `title` function.